### PR TITLE
Implementing _getOpenedPRs

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Implementing the `_getOpenedPRs` method that was introduced with screwdriver-cd/scm-base#37. 

This will look up the opened PRs against a specific repository. It will then return a payload of a few data points `name`, `prRef`.

Related to screwdriver-cd/screwdriver#417

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -303,6 +303,37 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Get a list of opened Pull Requests for a given repository
```
Comment (d2lam) [inline index.js]: Get list of objects which consist of opened PR names and its ref

File: test/index.test.js [hunk 2]
```diff
@@ -1324,10 +1317,94 @@ jobs:
...
+                    repo: 'repoName',
+                    state: 'open'
+                });
+            });
+        });
+
+        it('returns an empty list of pull requests', () => {
+            githubMock.pullRequests.getAll.yieldsAsync(null, []);
+
+            /* eslint-disable no-underscore-dangle */
```
Comment (minz1027) [inline test/index.test.js]: Can use `//eslint-disable-next-line no-underscore-dangle` to make it one line

File: test/index.test.js [hunk 3]
```diff
@@ -1324,10 +1317,94 @@ jobs:
...
+                assert.calledWith(githubMock.repos.getById, { id: '111' });
+                assert.calledWith(githubMock.pullRequests.getAll, {
+                    owner: 'repoOwner',
+                    repo: 'repoName',
+                    state: 'open'
+                });
+            });
+        });
+
+        it('returns an empty list of pull requests', () => {
```
Comment (minz1027) [inline test/index.test.js]: Why we need this test here? Looks like it is just testing if `map` function works with empty array

File: index.js [hunk 4]
```diff
@@ -303,6 +303,37 @@ class GithubScm extends Scm {
...
+     * @method _getOpenedPrs
+     * @param  {Object}      config
+     * @param  {String}      config.scmUri  The scmUri to get opened PRs from
+     * @param  {String}      config.token   The token used to authenticate with the SCM
+     * @return {Promise}
+     */
+    _getOpenedPrs(config) {
```
Comment (jerryzhang222) [inline index.js]: Should this be standardized with capital 'R' like in scm-base? ie getOpenedPRs

Comment (d2lam) [inline index.js]: Good catch! It actually needs to be `PRs`: https://github.com/screwdriver-cd/scm-base/blob/master/index.js#L296

File: test/index.test.js (hunk 2 above)
Comment (stjohnjohnson) [inline test/index.test.js]: Actually, you can set this up for the entire project via a eslint file in the test folder like this https://github.com/screwdriver-cd/executor-k8s/blob/master/test/.eslintrc.yaml

File: test/index.test.js (hunk 3 above)
Comment (FenrirUnbound) [inline test/index.test.js]: It was a "test for n", while the other unit test was the "test for n+2" scenario. It can be simplified
